contracts/bridge/BridgeTransferCommon.sol

```diff
@@ -127,11 +127,17 @@ contract BridgeTransfer is BridgeHandledRequests, BridgeFee, BridgeOperator {
         if (_requestedNonce > upperHandleNonce) {
             upperHandleNonce = _requestedNonce;
         }
-        for (i = lowerHandleNonce; i <= upperHandleNonce && handleNoncesToBlockNums[i] > 0; i++) { }
-        lowerHandleNonce = i;
-        if (i != 0) {
-            recoveryBlockNumber = handleNoncesToBlockNums[i-1];
+        for (i = lowerHandleNonce; i <= upperHandleNonce && handleNoncesToBlockNums[i] > 0; i++) {
+            recoveryBlockNumber = handleNoncesToBlockNums[i];
+            delete handleNoncesToBlockNums[i];
+            delete closedValueTransferVotes[i];
+            removeVoteData(VoteType.ValueTransfer, i);
         }
+        lowerHandleNonce = i;
+    }
+
+    function lowerHandleNonceCheck(uint64 _requestedNonce) internal {
+        require(lowerHandleNonce <= _requestedNonce, "removed vote");
```

Comment:
lowerHandleNonce is somewhat like a hint for the recovery. However, if there is huge nonce jump, there is gap between lowerHandleNonce and the largest nonce. In that case, can we check the duplicated vote safely?

Reply:
The nonces in gap between `lowerHandleNonce` and `largest nonce` can be prevented to vote again by `closedValueTransferVotes`.